app/models/manageiq/providers/vmware/infra_manager/inventory/collector/property_collector.rb

```diff
@@ -17,19 +17,24 @@ def ems_inventory_filter_spec(vim)
         RbVmomi::VIM.ObjectSpec(:obj => vim.serviceContent.rootFolder, :selectSet => root_folder_select_set),
         RbVmomi::VIM.ObjectSpec(:obj => vim.serviceContent.licenseManager),
       ],
-      :propSet   => ems_inventory_prop_set
+      :propSet   => ems_inventory_prop_set(vim.rev)
     )
   end
 
-  def ems_inventory_prop_set
-    property_set_from_file("ems_inventory")
+  def ems_inventory_prop_set(api_version)
+    property_set_from_file("ems_inventory", api_version)
   end
 
-  def property_set_from_file(file_name)
+  def property_set_from_file(file_name, api_version)
     engine_root = ManageIQ::Providers::Vmware::Engine.root
     hash = YAML.load_file(engine_root.join("config", "property_specs", "#{file_name}.yml"))
 
-    hash.collect do |type, props|
+    major_minor_version = api_version.split(".").take(2).join(".")
```

Comment:
VMware never adds properties in a patch version so we only need to check `major.minor`